src/wp-comments-post.php

```diff
@@ -22,6 +22,23 @@
 
 nocache_headers();
 
+// Check if user is not logged in and email exists in the comment form
+if ( ! is_user_logged_in() && isset( $_POST['email'] ) && ! empty( $_POST['email'] ) ) {
+	$email = sanitize_email( wp_unslash( $_POST['email'] ) );
+
+	// Check if the email belongs to a registered user
+	if ( email_exists( $email ) ) {
+		wp_die(
+			'<p>' . __( 'The email address you entered is already associated with a registered account. Please log in or use a different email address.' ) . '</p>',
```

Comment:
I think we should remove the " or use a different email address" part of the message to avoid encouraging registered users to just remove one character from the email to send their comment with having to log in. Yeah most people are lazy :)